ubxtool.cc

```diff
@@ -230,12 +230,12 @@ UBXMessage sendAndWaitForUBX(int fd, int seconds, basic_string_view<uint8_t> msg
       return waitForUBX(fd, seconds, ubxClass, ubxType);
     }
     catch(...) {
-      if(n==1)
-        throw;
-      cerr<<"Retransmit"<<endl;
-          
+      if(n > 1) {
+        cerr<<"Retransmit"<<endl;
+      }
     }
   }
+  throw;
```

Comment:
I don't understand this - what does this 'throw' do? Does this even compile? I may be dense or not reading it right but it looks weird.

Reply:
I moved it from line 234. It's what you're using to throw an exception when 3 attempts time-out. Moving it here to the end of the function however allows the compiler to see that the function either returns the ubxmessage, or throws, which silences the compile warning. The behaviour of the function is unchanged as the for() loop exits after a third unsuccessful attempt.

Reply:
My misunderstanding of catch(...) { throw; } behaviour in propagating the caught exception. Commit reverted.